### PR TITLE
initialize referal admins UI and set up handler to serve it

### PR DESCRIPTION
**Pull request by fuskovic**
What: This PR initializes the UI for the referral admin GUI and implements a handler to serve it on the newly opened private port for Marketing. This PR also includes handling of 404 http statuses.

Why: So marketing admins have a UI to interact with when performing CRUD operations for offers.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: satellite/marketing/marketingweb/server.go [hunk 1]
```diff
@@ -30,6 +37,16 @@ type Server struct {
 	listener net.Listener
 	server   http.Server
 }
+// The three pages contained in addPages are pages all templates require
+// This exists in order to limit handler verbosity
+func addPages(assets []string) ([]string){
+	d := dir + "pages/"
```
Comment (crawter) [inline satellite/marketing/marketingweb/server.go]: will root or rootPath be more obvious?

File: satellite/marketing/marketingweb/server.go [hunk 2]
```diff
@@ -5,17 +5,24 @@ package marketingweb
...
 	"golang.org/x/sync/errgroup"
+	"github.com/gorilla/mux"
 )
 
 // Error is satellite marketing error type
-var Error = errs.Class("satellite marketing error")
-
+var (
+	Error = errs.Class("satellite marketing error")
+	_, fp, _, _ = runtime.Caller(0)
```
Comment (rikysya) [inline satellite/marketing/marketingweb/server.go]: That file path will be bound to your local code location, that may not work for precomputed binaries. Also you set hard rule for path of your static files to be in folder where server.go was during compilation. But I am not sure if we care about making static files configurable this much

File: satellite/marketing/marketingweb/server.go (hunk 1 above)
Comment (fuskovic) [reply satellite/marketing/marketingweb/server.go]: I agree that rootPath is more obvious but I wanted to use a single letter variable for two reasons

1-limit verbosity on line 44. 
i.e. 
`{d+"base.html", d+"index.html,d+"banner.html"}` is more succinct then 
`{rootPath+"base.html", rootPath+"index.html",rootPath+"banner.html"}`

2-It's also proper convention according to https://talks.golang.org/2014/names.slide#6

File: satellite/marketing/marketingweb/pages/foModal.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        </div>
+                        <div class="form-group col-md-4">
+                            <label for="expiration">Credit Exp Date</label>
+                            <input type="date" class="form-control" id="expiration" placeholder="06/01/19">
+                        </div>
+                    </div>
+                    <div class="form-row">
+                        <div class="form-group col-md-4">
+                            <label for="give-credit">Give Credit</label>
+                            <input type="number" id="give-credit" value="$50">                        </div>
```
Comment (VinozzZ) [inline satellite/marketing/marketingweb/pages/foModal.html]: Nit pick: can you fix this format? The `</div>` is not aligned with the opening tag

File: satellite/marketing/marketingweb/pages/foModal.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                        <div class="form-group col-md-4">
+                            <label for="give-credit-exp">Give Credit Exp.</label>
+                            <input type="date" class="form-control" id="give-credit-exp">
+                        </div>
+                        <div class="form-group col-md-4">
+                            <label for="redeemable-capacity">Redeemable Capacity</label>
+                            <input type="text" class="form-control" id="redeemable-capacity">
+                        </div>
+                    </div>
+                    <div class="m-1 text-left">
```
Comment (VinozzZ) [inline satellite/marketing/marketingweb/pages/foModal.html]: I talked with Tikh yesterday about this. We decided to remove this status setting on create for now

File: satellite/marketing/marketingweb/server.go (hunk 1 above)
Comment (VinozzZ) [reply satellite/marketing/marketingweb/server.go]: How abour `rp`?

File: satellite/marketing/marketingweb/static/css/style.css [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.banner-txt{
+    font-family: 'Inter';
+    font-size: 18px;
+    color:white;
+}
+
+.offer-label{
+font-family: 'Inter';
```
Comment (VinozzZ) [inline satellite/marketing/marketingweb/static/css/style.css]: nit pick: the indentation here is a little off

Comment (fuskovic) [reply satellite/marketing/marketingweb/static/css/style.css]: done

File: satellite/marketing/marketingweb/pages/foModal.html (hunk 4 above)
Comment (fuskovic) [reply satellite/marketing/marketingweb/pages/foModal.html]: done

File: satellite/marketing/marketingweb/pages/foModal.html (hunk 3 above)
Comment (fuskovic) [reply satellite/marketing/marketingweb/pages/foModal.html]: done

File: satellite/marketing/marketingweb/server.go (hunk 1 above)
Comment (fuskovic) [reply satellite/marketing/marketingweb/server.go]: done

**Pull request merged**